transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -568,7 +565,9 @@ void epollInReady() {
                     pipeline.fireExceptionCaught(exception);
                 }
             } finally {
-                epollInFinally(config);
+                if (shouldStopReading(config)) {
+                    clearEpollIn();
+                }
```

Comment:
fwiw, this whole if block is quite repetitive so maybe it would be a good candidate for a new definition of `epollInFinally(Config)`.

Reply:
I was thinking about it but I found it more readable this way :)

Reply:
Fair enough reason. It's not a lot of code in any case.